lib/chef/azure/commands/enable.rb

```diff
@@ -81,18 +79,42 @@ def enable_chef
     @exit_code
   end
 
-  def install_chef_service
-    @exit_code, error_message = ChefService.new.install(@azure_plugin_log_location)
-    if @exit_code == 0
-      report_status_to_azure "chef-service installed", "success"
-    else
-      report_status_to_azure "chef-service install failed - #{error_message}", "error"
-    end
-    @exit_code
+  def load_chef_service_interval
+    value_from_json_file(handler_settings_file, 'runtimeSettings', '0', 'handlerSettings', 'publicSettings', 'chef_service_interval')
   end
 
   def enable_chef_service
-    @exit_code, error_message = ChefService.new.enable(@chef_extension_root, bootstrap_directory, @azure_plugin_log_location)
+    chef_service = ChefService.new
+    chef_service_interval = load_chef_service_interval
+
+    if chef_service_interval.empty?
+      @exit_code, error_message = chef_service.enable(
+        @chef_extension_root,
+        @azure_plugin_log_location
+      )
+    else
+      if chef_service_interval.to_i == 0

```

Comment:
I would move the logic inside `if chef_service_interval.to_i == 0` into the `ChefService` class and simply call. `chef_service.disable` and let the `disable` method of the `ChefService` edit the config and disable the service or cron.